ext/standard/var_unserializer.re

```diff
@@ -326,11 +326,6 @@ static zend_string *unserialize_str(const unsigned char **p, size_t len, size_t
 	zend_string *str = zend_string_safe_alloc(1, len, 0, 0);
 	unsigned char *end = *(unsigned char **)p+maxlen;
 
-	if (end < *p) {
-		zend_string_efree(str);
-		return NULL;
-	}
```

Comment:
This may not breaking anything that wasn't already broken because this was optimized out by the compiler, but I would prefer if we fixed this check instead of simply removing it

Reply:
The intention of the check seems to be to see if maxlen wraps around the address space.
I don't think that's possible, given that `maxlen = max - YYCURSOR;` and the fact that `max` is the pointer to the end of the input string.
So I'm not sure what you want me to replace this check with?

Reply:
You are right. I was blindly assuming that maxlen was possibly user-controlled. Looking at the history this check made more sense at some point, but now it can be removed.

Reply:
This is out of scope, but we could pass `max` directly to this function, maybe via UNSERIALIZE_PASSTHRU/UNSERIALIZE_PARAMETER.